network/src/lib.rs

```diff
@@ -447,6 +457,7 @@ impl PolkadotProtocol {
 		if info.validator_keys.as_slice().is_empty() {
 			ctx.report_peer(who, cost::UNEXPECTED_MESSAGE);
 		} else {
+			ctx.report_peer(who.clone(), cost::EXPECTED_MESSAGE);
```

Comment:
this message is allowed but I don't know if we should bump reputation every time we receive it, because there is no limit on this. I think the `EXPECTED_MESSAGE` variant should be used only on successful responses to requests.

for collation roles, we should give a benefit on the first time only